main.rb

```diff
@@ -0,0 +1,140 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    result = []
+    my_each { |x| result << x if yield(x) }
+    result
+  end
+
+  # rubocop: disable Style/CaseEquality:
+
+  def my_all?(args = nil)
+    my_each do |i|
+      if block_given?
+        return false unless yield(i)
+      elsif args.nil?
+        return false unless i
+      else
+        return false unless args === i
+      end
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    my_each do |x|
+      if block_given?
+        return true if yield(x)
+      elsif args.nil?
+        return true if x === args
+      elsif x
+        return true
+      end
+    end
+    false
+  end

```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

-  When no block or argument is given returns false if at least one of the collection is not true
```ruby
[false, false, nil].my_any? == [false, false, nil].any? # => false
```
Yours (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/84203069-0b3aa400-aa66-11ea-82d0-13c52ab4b373.png)

Real any? (correct)
![image](https://user-images.githubusercontent.com/52209067/84203096-18f02980-aa66-11ea-87a5-41d7f3512340.png)

- When a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer

```Ruby
['1', '2', '5', false].my_any?(Integer) == ['1', '2', '5', false].any?(Integer) # => false
```
Yours (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/84203398-9b78e900-aa66-11ea-86af-7621d1684ff3.png)

Real any? (correct)
![image](https://user-images.githubusercontent.com/52209067/84203430-a7fd4180-aa66-11ea-87bb-f9b8154a0864.png)

- when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
['lorem','ipsum', 'dolo'].my_any?(/z/) == ['lorem','ipsum', 'dolo'].any?(/z/) # => false
```
Yours (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/84204069-ca438f00-aa67-11ea-9af8-a69880613a4f.png)

Real any? (correct)
![image](https://user-images.githubusercontent.com/52209067/84204098-d6c7e780-aa67-11ea-8cd9-195e96ae927b.png)

- When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
['lorem','ipsum', 'dolo'].my_any?('cat') == ['lorem','ipsum', 'dolo'].any?('cat') # => false
```
Yours (wrong answer)
![image](https://user-images.githubusercontent.com/52209067/84205873-45f30b00-aa6b-11ea-868b-712204c11d1d.png)

Real any? (correct)
![image](https://user-images.githubusercontent.com/52209067/84205896-51decd00-aa6b-11ea-974a-63e639f3b869.png)